File: 53. Maximum Subarray.c
int maxSubArray(int* nums, int numsSize) {
    int s1 = 0, s2 = 0;
    int i;
    int maxnums = nums[0];
    
    for (i = 0; i < numsSize; i++){
        if (*(nums+i) + s1 < 0)
            s1 = 0;
        else
            s1 = *(nums+i) + s1;
        if (s1 > s2)
            s2 = s1;
        if (maxnums < *(nums+i))
            maxnums = *(nums+i);
    }
    if (s2 == 0)
        return maxnums;
    else
        return s2;
}

File: 053. Maximum Subarray/53. Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        https://discuss.leetcode.com/topic/69415/python-solution-o-n-beat-99-94-other-solutions
        """
        s1 = s2 = 0
        for num in nums:
            s1 = max(0, s1 + num)
            s2 = max(s1, s2)
        return max(nums) if not s2 else s2
